modules/dice/roll.js

```diff
@@ -272,10 +272,17 @@ export class RollFFG extends Roll {
       addedResults: this.addedResults,
       publicRoll: !chatOptions.isPrivate,
     };
-    if (chatData.data.hasOwnProperty('data') && chatData.data.data.adjusteditemmodifier === undefined) {
+    if (chatData.data.hasOwnProperty('data') && (chatData.data.data.adjusteditemmodifier === undefined || chatData.data.data.adjusteditemmodifier.length === 0)) {
       // extended metadata is missing, lookup the actor ID so we can embed it for future lookups
       let candidate_actors = game.actors.filter(actor => actor.items.filter(item => item.id === chatData.data._id).length > 0);
       if (candidate_actors.length > 0) {
+        let test_item = game.actors.get(candidate_actors[0].id).items.get(chatData.data._id);
+        // for whatever reason, sometimes the item we read doesn't have modifiers even though the chat item does
+        // check if this is the case and correct it if it is
+        if (test_item.data?.data?.itemmodifier.length === 0 && chatData.data?.data?.itemmodifier) {
+          // there aren't any modifiers on the object, try copying the temp object to it so the link works
+          test_item.data.data.itemmodifier = chatData.data.data.itemmodifier;
```

Comment:
We're not calling a document update here so this change is clientside only, hence I'm not too worried, but...

This feels like a weird way of handling it in scenarios like;
1) I roll an item to chat
2) I edit that item, removing various things
3) The chat message has old data now, but is still in the chat log
4) Whenever the chatlog is reloaded, such as a page refresh, the chatlog updates my clientside item to have the old data

Reply:
That isn't the case I was addressing with this. When I migrated my old world (from Foundry 0.7.10), the item (when looked up) didn't have any item modifiers, but the chat data version of the item did.  Rather than find the source of the lack of information, I just hacked it together so the links would work.

Reply:
Maybe we should use `test_item.data?.data?.itemmodifier === undefined` rather than the length check in that case, as that's less likely catch the legitimate scenarios alongside the actual buggy ones.

Reply:
It wasn't undefined, it was just an empty list.

Reply:
I can wrap it in a try/except block to avoid breaking things if the properties are missing - would that make you more comfortable with it?

Reply:
Done. I also removed the extra console log. Let me know if you want further changes.

Reply:
My concern isn't really around the null handling, more that updating item data from chat data is a bit of an anti-pattern, unless we have a way of being 100% sure the conditions will only fire with a bug.

Right now I can see the conditions being truthy any time an item has had its modifiers added / removed at any point after it's been rolled to chat. So any old chat message in the chat log will revert its clientside data back to the state it was in when that message was sent (which could be months ago in an old world file)...

The result of that would be I seem unable to update my item's modifiers, I keep removing them and then every time I load the page fresh and the chat log renders its messages, my item reverts to some old state.

Reply:
I could register a setting and only enable this functionality if it's turned on (and default it to off)?

Basically, I want item quality hyperlinking to work for ported worlds without having to debug why the items are missing data in the first place, since this is likely to be the last release for this system.

Reply:
Fair enough, feature flag it behind a setting and I'll merge

Reply:
Donezo.